src/core/thread/mle_router.cpp

```diff
@@ -2072,6 +2082,25 @@ otError MleRouter::UpdateChildAddresses(const Message &aMessage, uint16_t aOffse
 
             memcpy(&address, context.mPrefix, BitVectorBytes(context.mPrefixLength));
             address.SetIid(entry.GetIid());
+
+#if OPENTHREAD_CONFIG_TMF_PROXY_DUA_ENABLE
```

Comment:
Can we move most logic here to `DuaManager` so that we make minimal change to MleRouter ?

Reply:
new design will only require to check Added or Removed, pls take another look when new patch is submitted.

Reply:
@simonlingoogle 
There is no copy for child's DUA in `DuaManager`, and the information will be refreshed with AR TLV per Child Update Request, I think it is a proper place to check the child's DUA changes along with `UpdateChildAddresses()` here as the history information will be lost afterwards.

Would you please share more specific suggestion on possible the other way to minimize the change here?

Reply:
DuaManager needs to know if DUA is removed or added or changed. 
One way I can think of is like this:

```c++
// before update Child addresses
bool oldHasDua = aChild.GetDomainUnicastAddress() != nullptr;
// After update Child addresses
Get<DuaManager>().UpdateChildDomainUnicastAddress(aChild, oldHasDua);
```

We can then find for the new Dua in `UpdateChildDomainUnicastAddress` itself. 
So we can minimize changes to `MleRouter` by only collecting the minimal information we need before updating Child addresses and pass it to `DuaManager` for further logic.

Reply:
Another iteration through Child's IP addresses should be cheap since it's just iteration of indexes.

We can retrieve the old Dua and pass it to DuaManager like below?

```c++
// before update Child addresses
bool oldHasDua = (aChild.GetDomainUnicastAddress(oldDua) == OT_ERROR_NONE);...

// After update Child addresses
Get<DuaManager>().UpdateChildDomainUnicastAddress(aChild, oldHasDua, oldDua);
```

Reply:
Yes, it is an option which takes the same process style like proxy MLR, however I still think it is more naturally to put it here:

Different from proxy MLR which requires iterating through the MAs the child registered previously, and MAs per other child registered, plus MAs registered by the parent itself, for proxy DUA, only need to simply compare with the one previously registered if any.

Reply:
Fine to keep it as is.